tests/test_simulation_program.py

```
@@ -0,0 +1,983 @@
+    DigitalSegment,
+    DigitalSimParams,
+    Hamiltonian,
+    NoiseModel,
+    Observable,
+    SimulationProgram,
+    Simulator,
+    State,
+)
+from mqt.yaqs.core.data_structures.mpo import MPO
+from mqt.yaqs.core.data_structures.simulation_program import (
+    _compile_program,  # ruff: ignore[import-private-name]  # validate private compiler invariant
```

Ruff (0.16.1)

[warning] Invalid rule code in suppression: import-private-name

Enable `lint.preview` to use rule names

(RUF102)